mypy/fastparse.py

```diff
@@ -184,18 +187,28 @@ def parse_type_comment(type_comment: str,
                        line: int,
                        errors: Optional[Errors],
                        assume_str_is_unicode: bool = True,
-                       ) -> Optional[Type]:
+                       ) -> Tuple[bool, Optional[Type]]:
     try:
         typ = ast3_parse(type_comment, '<type_comment>', 'eval')
     except SyntaxError as e:
         if errors is not None:
             errors.report(line, e.offset, TYPE_COMMENT_SYNTAX_ERROR, blocker=True)
-            return None
+            return False, None
         else:
             raise
     else:
+        # We need to handle the case where an
+        # extra # type: ignore is in a comment, like this line.
+        # Therefore we split on comment delineations and check
+        # only the first possibility for a type ignore.
+        parts = type_comment.split('#')
+        if len(parts) > 1:
+            _, possible_ignore, *rest = parts
+        else:
+            possible_ignore = type_comment
```

Comment:
You don't need to check the type comment itself for looking like `ignore` -- typed_ast (or ast in 3.8) has already done that.

I also wonder if you could omit splitting on `#` at all and just use a slightly more complex regex. I think something like this would do it:
```
extra_ignore = re.match(r'[^#]*#\s*type:\s*ignore\s*($|#)', type_comment) is not None
```